File: WareHouse/32_DefaultOrganization_20241226051308/gamelogic.java
import javax.swing.JButton;
public class GameLogic {
    public static final int BOARD_SIZE = 8;
    private char[][] gameBoard;
    private int currentPlayer;
    private boolean gameOver;
    private JButton[][] gameBoardButtons;
    public GameLogic(JButton[][] gameBoardButtons) {
        this.gameBoardButtons = gameBoardButtons;
        gameBoard = new char[BOARD_SIZE][BOARD_SIZE];
        initializeGameBoard();
        currentPlayer = 1;
        gameOver = false;
    }
    private void initializeGameBoard() {
        // Initialize the game board with empty spaces
        for (int i = 0; i < BOARD_SIZE; i++) {
            for (int j = 0; j < BOARD_SIZE; j++) {
                gameBoard[i][j] = ' ';
            }
        }
    }
    /**
     * This method makes a move on the game board.
     *
     * @param row The row index of the move.
     * @param col The column index of the move.
     */
    public void makeMove(int row, int col) {
        // Check if the move is valid
        if (!gameOver && isValidMove(row, col)) {
            // Update the game board with the move
            gameBoard[row][col] = (currentPlayer == 1) ? 'X' : 'O';
            gameBoardButtons[row][col].setText(String.valueOf(gameBoard[row][col]));
            currentPlayer = (currentPlayer == 1) ? 2 : 1;
            if (isGameOver()) {
                gameOver = true;
                System.out.println("Player " + ((currentPlayer == 1) ? 2 : 1) + " wins!");
            } else if (isGameDraw()) {
                gameOver = true;
                System.out.println("The game is a draw!");
            }
        }
    }
    private boolean isValidMove(int row, int col) {
        // Add your logic to check if the move is valid
        // For example, check if the specified position is empty or within the board boundaries
        // Return true if the move is valid, false otherwise
        if (row >= 0 && row < BOARD_SIZE && col >= 0 && col < BOARD_SIZE && gameBoard[row][col] == ' ') {
            return true;
        }
        return false;
    }
    private boolean isGameOver() {
        // Check for win conditions
        for (int row = 0; row < BOARD_SIZE; row++) {
            if (gameBoard[row][0] != ' ' && gameBoard[row][0] == gameBoard[row][1] && gameBoard[row][0] == gameBoard[row][2]) {
                return true;
            }
        }
        for (int col = 0; col < BOARD_SIZE; col++) {
            if (gameBoard[0][col] != ' ' && gameBoard[0][col] == gameBoard[1][col] && gameBoard[0][col] == gameBoard[2][col]) {
                return true;
            }
        }
        if (gameBoard[0][0] != ' ' && gameBoard[0][0] == gameBoard[1][1] && gameBoard[0][0] == gameBoard[2][2]) {
            return true;
        }
        if (gameBoard[0][2] != ' ' && gameBoard[0][2] == gameBoard[1][1] && gameBoard[0][2] == gameBoard[2][0]) {
            return true;
        }
        return false;
    }
    private boolean isGameDraw() {
        // Check if all positions on the game board are filled
        for (int row = 0; row < BOARD_SIZE; row++) {
            for (int col = 0; col < BOARD_SIZE; col++) {
                if (gameBoard[row][col] == ' ') {
                    return false;
                }
            }
        }
        return true;
    }
    /**
     * This method returns the current game board.
     *
     * @return The current game board.
     */
    public char[][] getGameBoard() {
        return gameBoard;
    }
}

File: WareHouse/C_278__20250511050407/game_logic.py
'''
Contains the game logic and rules for the Gomoku game.
'''
class GameLogic:
    def __init__(self, board_size):
        '''
        Initializes the game logic with a specified board size.
        Parameters:
        board_size (int): The size of the board (number of rows and columns).
        '''
        self.board_size = board_size
        self.board = [['' for _ in range(board_size)] for _ in range(board_size)]
    def place_stone(self, x, y, player):
        '''
        Places a stone on the board at the specified coordinates if the move is valid.
        Parameters:
        x (int): The row index where the stone is to be placed.
        y (int): The column index where the stone is to be placed.
        player (str): The player making the move ('black' or 'white').
        Returns:
        bool: True if the stone was placed successfully, False otherwise.
        '''
        if 0 <= x < self.board_size and 0 <= y < self.board_size:  # Check bounds
            if self.board[x][y] == '':
                self.board[x][y] = player
                return True
        return False
    def is_winner(self, player):
        '''
        Checks if the specified player has won the game.
        Parameters:
        player (str): The player to check for a win ('black' or 'white').
        Returns:
        bool: True if the player has won, False otherwise.
        '''
        for x in range(self.board_size):
            for y in range(self.board_size):
                if self.board[x][y] == player:
                    if self.check_direction(x, y, player):
                        return True
        return False
    def check_direction(self, x, y, player):
        '''
        Checks all possible winning directions from a given position.
        Parameters:
        x (int): The row index of the starting position.
        y (int): The column index of the starting position.
        player (str): The player to check for a win ('black' or 'white').
        Returns:
        bool: True if there is a winning line, False otherwise.
        '''
        directions = [(1, 0), (0, 1), (1, 1), (1, -1)]
        for dx, dy in directions:
            count = 1
            # Check in the positive direction
            for step in range(1, 5):
                nx, ny = x + dx * step, y + dy * step
                if 0 <= nx < self.board_size and 0 <= ny < self.board_size and self.board[nx][ny] == player:
                    count += 1
                else:
                    break
            # Check in the negative direction
            for step in range(1, 5):
                nx, ny = x - dx * step, y - dy * step
                if 0 <= nx < self.board_size and 0 <= ny < self.board_size and self.board[nx][ny] == player:
                    count += 1
                else:
                    break
            if count >= 5:
                return True
        return False
    def is_draw(self):
        '''
        Checks if the game is a draw (i.e., there are no valid moves left).
        Returns:
        bool: True if the game is a draw, False otherwise.
        '''
        # Check if the board is full
        for row in self.board:
            if '' in row:
                return False
        # If the board is full, check for winners
        return not self.is_winner('black') and not self.is_winner('white')
    def reset(self):
        '''
        Resets the game board to its initial empty state.
        '''
        self.board = [['' for _ in range(self.board_size)] for _ in range(self.board_size)]